Sources/SwiftParserDiagnostics/ParserDiagnosticMessages.swift

```diff
@@ -305,6 +296,30 @@ public struct CannotParseVersionTuple: ParserError {
   }
 }
 
+public struct DeinitializerSignatureError: ParserError {
+  public let name: TokenSyntax?
+  public let params: FunctionParameterClauseSyntax?
+  public let returnClause: ReturnClauseSyntax?
+
+  public var message: String {
+    var descriptions: [String] = []
+
+    if name != nil {
+      descriptions += ["name"]
+    }
+
+    if params != nil {
+      descriptions += ["parameters"]
+    }
+
+    if returnClause != nil {
+      descriptions += ["return clause"]
+    }
```

Comment:
Another minor suggestion: How about using the `append` method instead of creating new arrays with single elements? The code works just fine as is, but using `append` could make it feel a bit more like Swift in its natural habitat. For example:
```suggestion
    if name != nil {
      descriptions.append("name")
    }

    if params != nil {
      descriptions.append("parameters")
    }

    if returnClause != nil {
      descriptions.append("return clause")
    }
```